metau-capstone/View Controllers/PreviewViewController.m

```diff
@@ -59,21 +85,53 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
     // Pass the selected object to the new view controller.
 }
 */
+- (void)frontTextFieldDidChange: (UIButton*)sender {
+    PreviewCard *card = self.previewCards[self.currentCellPath.row];
+    card.frontText = self.frontTextField.text;
+    [self.previewCarousel reloadItemsAtIndexPaths:@[self.currentCellPath]];
+}
 
+- (void)backTextFieldDidChange: (UIButton*)sender {
+    PreviewCard *card = self.previewCards[self.currentCellPath.row];
+    card.backText = self.backTextField.text;
+    [self.previewCarousel reloadItemsAtIndexPaths:@[self.currentCellPath]];
+}
 
 - (nonnull __kindof UICollectionViewCell *)collectionView:(nonnull UICollectionView *)collectionView cellForItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
     PreviewCell *cell = [self.previewCarousel dequeueReusableCellWithReuseIdentifier:@"PreviewCell" forIndexPath:indexPath];
     PreviewCard *card = self.previewCards[indexPath.row];
     [cell createCardBothSides:CGRectMake(10, 70, 270, 162) withFront:card.frontText withBack:card.backText];
+    [self setActionForButton:cell.editButton withTag:indexPath.row withAction:@selector(didTapEdit:)];
```

Comment:
This is not a good practice, often what you want to do is the following:
1. defined 2 delegate functions in `PreviewCell` that will be invoked when cell button is clicked
2. make `PreviewViewController` conform to PreviewCellDelegate
3. create a new function in `PreviewCell` configureWithPreviewCard:withDelegate
4. inside this function `collectionView: cellForItemAtIndexPath` you invoke `[cell configureWithPreviewCard:card withDelegate:self];`
5. inside the `PreviewCell` selector for either the edit or select button, invoke `[self.delegate onEditButtonTappedForCard: self.card];`